sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonParser.scala

```diff
@@ -90,6 +91,10 @@ class JacksonParser(
         // in such an array as a row, this case is possible.
         if (array.numElements() == 0) {
           Nil
+        } else if (array.numElements() > 1 && !explodeArray) {
```

Comment:
shall we forbid array as struct type completely for `from_json`?

BTW I think this feature is not needed now, since we can directly read top-level json arrays as array type. We may still need to keep it when reading json files, but I think it's ok to drop it in `from_json`.

Reply:
the new parameter can be named as `allowArrayAsStructs`

Reply:
> ... but I think it's ok to drop it in from_json
> ... the new parameter can be named as allowArrayAsStructs

Is it ok if I'll do:

```
case START_ARRAY if allowArrayAsStructs =>
```